### PR TITLE
Feature/customerloyalty credential

### PR DESCRIPTION
**Pull request by ottonomy**
Prerequisite: Merge PR #21 to add the context to the repository.

Adds a `CustomerLoyaltyCredential` example to the repository. 

> Retail establishments often offer a [loyalty program](https://en.wikipedia.org/wiki/Loyalty_program) to encourage customers to make repeat purchases. A loyalty program typically involves a customer signing up for a program and receiving a loyalty card. The customer presents the loyalty card at the point of sale to receive discounts or other benefits. The loyalty card is typically a plastic card with a barcode or magnetic stripe that can be scanned at the point of sale. The loyalty card is associated with a customer account identifier, so that the store can track the customer's purchases and apply rewards.
>
> A `CustomerLoyaltyCredential` is a digital representation of a loyalty card. It can be presented by a customer to a retailer to receive discounts or other benefits. The credential asserts the customer account identifier and optionally which location the card is associated with. The credential is signed and verified by the retailer.

### CODE REVIEW

**Review by dlongley — CHANGES_REQUESTED** [review]

**Comment by davidlehn** [comment]
@ottonomy:
- The PR base commit should be adjusted to the branch for the #21 PR.
- There are context comments here that should likely be in #21, so I didn't change the base in case it would drop those.

File: contexts/customer-loyalty/v1.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "@context": {
+    "@version": 1.1,
```
[the remark this answers is not in the document]
Comment (ottonomy) [reply contexts/customer-loyalty/v1.json:3]: Resolved in #21

**Comment by ottonomy** [comment]
Status now after I came back from vacation:
* This PR is in draft state and is not planned to be reviewed or merged until after #21 is merged, at which point this will switch to ready-for-review, which is why the commit that adds context from that branch is also present in this branch. It is a prerequisite to the commits here making any sense, they are not independent changes. Confirmed this strategy via meeting earlier today. 
* Context comments left in this PR have been resolved on the branch for #21, which is ready for review.
* After #21 is merged, this will be rebased to resolve changes to `CHANGELOG.md`.

**Review by dlongley — APPROVED** [review]
Thanks!

File: credentials/customer-loyalty/credential.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    "customerLoyaltyCard": {
+      "type": "CustomerLoyaltyCard",
+      "branchCode": "461",
+      "identifier": "9712323901"
+    }
+  },
+  "issuanceDate": "2023-06-12T00:00:00Z",
+  "issuer": {
+    "id": "did:key:z6MkrHKzgsahxBLyNAbLQyB1pcWNYC9GmywiWPgkrvntAZcj",
+    "image": "https://playground.chapi.io/examples/retail-coupon/logo.png",
```
Comment (davidlehn) [inline credentials/customer-loyalty/credential.json:21]: Use vcplayground.org:
```suggestion
    "image": "https://vcplayground.org/examples/retail-coupon/logo.png",
```
I'd consider copying the logo image over to customer-loyalty so the example is self contained and resistant to changes in the retail-coupon example.

File: credentials/customer-loyalty/credential.json [hunk 3]
```diff
@@ -0,0 +1,25 @@
+{
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://contexts.vcplayground.org/examples/customer-loyalty/v1.json"
+  ],
+  "type": ["VerifiableCredential", "CustomerLoyaltyCredential"],
+  "description": "Earn Big Rewards at Big Retail Store by presenting your loyalty card!",
+  "image": "https://playground.chapi.io/examples/customer-loyalty/image.png",
```
Comment (davidlehn) [inline credentials/customer-loyalty/credential.json:8]: Use vcplayground.org:
```suggestion
  "image": "https://vcplayground.org/examples/customer-loyalty/image.png",
```

**Comment by BigBlueHat** [comment]
@ottonomy this looks nearly done. Could you consolidate the images and update the VC Playground URLs?

Thanks!

**Comment by gannan08** [comment]
@BigBlueHat I will takeover this PR.

**Pull request merged**